### PR TITLE
Install EKSA components on bootstrap task for controller via CLI

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
Install EKSA components on bootstrap task for controller via CLI

*Testing (if applicable):*
- unit tests

*Documentation added/planned (if applicable):*
N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/dependencies/factory.go [hunk 1]
```diff
@@ -1076,16 +1077,14 @@ func (f *Factory) WithClusterManager(clusterConfig *v1alpha1.Cluster, timeoutOpt
...
-
 		f.dependencies.ClusterManager = clustermanager.New(
 			f.dependencies.UnAuthKubeClient,
 			client,
 			f.dependencies.Networking,
 			f.dependencies.Writer,
 			f.dependencies.DignosticCollectorFactory,
 			f.dependencies.AwsIamAuth,
-			installer,
+			f.dependencies.EksaInstaller,
```
Comment (g-gaston) [inline pkg/dependencies/factory.go:1087]: need to add `WithEKSAInstaller` to clusters manager deps

File: pkg/workflows/interfaces/interfaces.go [hunk 2]
```diff
@@ -99,3 +101,8 @@ type ClusterCreator interface {
 	Run(ctx context.Context, spec *cluster.Spec, managementCluster types.Cluster) error
 	CreateSync(ctx context.Context, spec *cluster.Spec, managementCluster *types.Cluster) (*types.Cluster, error)
 }
+
+// EksaInstaller exposes the EKSA installer methods.
+type EksaInstaller interface {
```
Comment (g-gaston) [inline pkg/workflows/interfaces/interfaces.go:106]:
```suggestion
// EksaInstaller installs the EKS-A controllers and CRDs.
type EksaInstaller interface {
```

File: pkg/workflows/management/create_install_eksa.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+func installEKSAComponents(ctx context.Context, commandContext *task.CommandContext, targetCluster *types.Cluster) error {
+	logger.Info("Installing EKS-D components")
+	err := commandContext.EksdInstaller.InstallEksdCRDs(ctx, commandContext.ClusterSpec, targetCluster)
+	if err != nil {
+		commandContext.SetError(err)
+		return err
+	}
+
+	logger.Info("Installing EKS-A custom components (CRD and controller)")
+	err = commandContext.ClusterManager.InstallCustomComponents(ctx, commandContext.ClusterSpec, targetCluster, commandContext.Provider)
```
Comment (g-gaston) [inline pkg/workflows/management/create_install_eksa.go:48]: Why is this going through cluster manager instead of directly using the eksa installer?

Comment (mitalipaygude) [reply pkg/workflows/management/create_install_eksa.go]: Updated it to use the EKSA installer instead.

**Review by g-gaston — COMMENTED** [review]
/lgtm

**Pull request merged**